test/unit/math/prim/prob/categorical_log_test.cpp

```diff
@@ -6,12 +6,12 @@ TEST(ProbDistributionsCategorical, log_matches_lpmf) {
   Eigen::Matrix<double, Eigen::Dynamic, 1> theta(3, 1);
   theta << 0.3, 0.5, 0.2;
 
-  EXPECT_FLOAT_EQ((stan::math::categorical_lpmf<true, double>(1, theta)),
-                  (stan::math::categorical_log<true, double>(1, theta)));
-  EXPECT_FLOAT_EQ((stan::math::categorical_lpmf<false, double>(1, theta)),
-                  (stan::math::categorical_log<false, double>(1, theta)));
-  EXPECT_FLOAT_EQ((stan::math::categorical_lpmf<double>(1, theta)),
-                  (stan::math::categorical_log<double>(1, theta)));
+  EXPECT_FLOAT_EQ((stan::math::categorical_lpmf<true>(1, theta)),
+                  (stan::math::categorical_log<true>(1, theta)));
+  EXPECT_FLOAT_EQ((stan::math::categorical_lpmf<false>(1, theta)),
+                  (stan::math::categorical_log<false>(1, theta)));
+  EXPECT_FLOAT_EQ((stan::math::categorical_lpmf<>(1, theta)),
```

Comment:
Is this removal necessary or just removing to cleanup code? It might be nicer to have these without the empty `<>`

Reply:
Now the template arguments have different meaning. So instantiating it with double would mean theta is double. Also now explicit template arguments are  never necessary. I can also delete the empty angle brackets.

Reply:
Ok thanks. The cleanup is not really necessary, was more wondering if it was mandatory or something. Good to go.

Reply:
It turns out some of these tests were duplicates. Fixed.